src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,719 @@
+    code_ += double_spacing + "}";
+    code_ += spacing + "}";
+    code_ += spacing + "return nil";
+    code_ += "\t}";
+  }
+
+  std::string GenOffset() { return "let o = {{ACCESS}}.offset({{OFFSET}}); "; }
+
+  std::string GenReaderMainBody(std::string optional = "") {
+    return "\tpublic var {{VALUENAME}}: {{VALUETYPE}}" + optional + " { ";
+  }
+
+  std::string GenReader(std::string type, std::string at = "{{OFFSET}}") {
+    return "{{ACCESS}}.readBuffer(of: {{" + type + "}}.self, at: " + at + ")";
+  }
+
+  std::string GenConstructor(
+      std::string offset = "{{ACCESS}}.postion + {{OFFSET}}") {
+    return "{{VALUETYPE}}({{ACCESS}}.bb, o: " + offset + ") }";
+  }
+
+  std::string GenMutate(std::string offset, std::string get_offset = "",
+                        bool isRaw = false) {
+    return "\tpublic func mutate({{VALUENAME}}: {{VALUETYPE}}) -> Bool {" +
+           get_offset + " return {{ACCESS}}.mutate({{VALUENAME}}" +
+           (isRaw ? ".rawValue" : "") + ", index: " + offset + ") }";
+  }
+
+  std::string GenMutateArray() {
+    return "\tpublic func mutate({{VALUENAME}}: {{VALUETYPE}}, at index: "
+           "Int32) -> Bool { " +
+           GenOffset() +
+           "return {{ACCESS}}.directMutate({{VALUENAME}}, index: "
+           "{{ACCESS}}.vector(at: o) + index * {{SIZE}}) }";
+  }
+
+  std::string GenEnumBaseContructor() {
+    return "{{VALUETYPE}}(rawValue: {{CONSTANT}})!";
+  }
+
+  std::string GenEnumConstructor(std::string at = "{{OFFSET}}") {
+    return "{{VALUETYPE}}(rawValue: " + GenReader("BASEVALUE", at) + ") ";
+  }
+
+  std::string GenType(const Type &type) const {
+    return IsScalar(type.base_type)
+               ? GenTypeBasic(type)
+               : (IsArray(type) ? GenType(type.VectorType())
+                                : GenTypePointer(type));
+  }
+
+  std::string GenTypePointer(const Type &type) const {
+    switch (type.base_type) {
+      case BASE_TYPE_STRING: return "String";
+      case BASE_TYPE_VECTOR: return GenType(type.VectorType());
+      case BASE_TYPE_STRUCT: return WrapInNameSpace(*type.struct_def);
+      case BASE_TYPE_UNION:
+      default: return "FlatBufferObject";
+    }
+  }
+
+  std::string GenTypeBasic(const Type &type) const {
+    return GenTypeBasic(type, true);
+  }
+
+  std::string GenTypeBasic(const Type &type, bool can_override) const {
+    // clang-format off
+    static const char * const swift_type[] = {
+      #define FLATBUFFERS_TD(ENUM, IDLTYPE, \
+              CTYPE, JTYPE, GTYPE, NTYPE, PTYPE, RTYPE, KTYPE, STYPE) \
+        #STYPE,
+        FLATBUFFERS_GEN_TYPES(FLATBUFFERS_TD)
+      #undef FLATBUFFERS_TD
+    };
+    // clang-format on
+    if (can_override) {
+      if (type.enum_def)
+        return WrapInNameSpace(type.enum_def->defined_namespace,
+                               Name(*type.enum_def));
+      if (type.base_type == BASE_TYPE_BOOL) return "Bool";
+    }
+    return swift_type[type.base_type];
```

Comment:
V2006 Implicit type conversion from enum type to integer type. idl_gen_swift.cpp 633

Better to use explicit static_cast here.